build/bind_artifacts.jl

```diff
@@ -25,37 +23,24 @@ function sha256sum(tarball_path)
     end
 end
 
-function get_release_asset_urls()
-    io = IOBuffer()
-    Downloads.download(ASSETS_URL, io)
-    json = JSON3.read(String(take!(io)))
-    return unescapeuri.(get.(json[:assets], :browser_download_url))
-end
-
 if abspath(PROGRAM_FILE) == @__FILE__
-    @info "Fetching assets for $TAG"
-    artifacts_urls = get_release_asset_urls()
+    # Start with a clean Artifacts.toml so that unsupported platforms are removed
+    isfile(ARTIFACTS_TOML) && rm(ARTIFACTS_TOML)
 
-    for artifact_url in artifacts_urls
-        @info "Adding artifact for $(basename(artifact_url))"
-        artifact_path = joinpath(DIR, basename(artifact_url))
-        Downloads.download(artifact_url, artifact_path)
-
-        m = match(TARBALL_REGEX, basename(artifact_path))
-        if isnothing(m)
-            throw(ArgumentError("Could not parse host triplet from $(basename(artifact_path))"))
-        end
+    for platform in REQUIRED_PLATFORMS
+        artifact_name = gen_artifact_filename(; tag=TAG, platform)
+        artifact_url = gen_artifact_url(; repo_url=REPO_HTTPS_URL, tag=TAG,
+                                        filename=artifact_name)
 
-        platform_triplet = m[:triplet]
-        julia_version = m[:julia_version]
-        platform_str = "$platform_triplet-julia_version+$julia_version"
-        platform = parse(BinaryPlatforms.Platform, platform_str)
+        @info "Dowloading artifact $artifact_url"
+        artifact_path = joinpath(TARBALL_DIR, artifact_name)
```

Comment:
Previously we downloaded files to a temporary directory instead of the `TARBALL_DIR`. This will override any locally generated files but that seems fine.